File: LeetCode/Medium/102 - Binary Tree Level Order Traversal/102 - Binary Tree Level Order Traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrder = function(root) {
    if (!root) return [];
  
    const queue = [], ret = [];
    queue.push(root);
    while (queue.length) {
        const len = queue.length, curLevel = [];
        for (let i = 0; i < len; ++i) {
            const curNode = queue.shift();
            curLevel.push(curNode.val);
            if (curNode.left) queue.push(curNode.left);
            if (curNode.right) queue.push(curNode.right);
        }
        ret.push(curLevel);
    }
    
    return ret;
};

File: LeetCode/Medium/102 - Binary Tree Level Order Traversal/102 - Binary Tree Level Order Traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        if (!root) return {};
        vector<vector<int>> ret;
        queue<TreeNode*> qu;
        qu.push(root);
        while (!qu.empty()) {
            int n = qu.size();
            vector<int> vc;
            while (n--) {
                vc.push_back(qu.front()->val);
                if (qu.front()->left) qu.push(qu.front()->left);
                if (qu.front()->right) qu.push(qu.front()->right);
                qu.pop();
            }
            ret.push_back(vc);
        }
        return ret;
    }
};